src/or/hs_common.h

```diff
@@ -141,6 +141,12 @@ typedef enum {
   RSAE_OKAY        = 0   /**< Service added as expected */
 } hs_service_add_ephemeral_status_t;
 
+/* Type of client authorization. */
+typedef enum {
+  HS_CLIENT_AUTH_TYPE_NULL = 0,
+  HS_CLIENT_AUTH_TYPE_BASIC = 1,
```

Comment:
We had so much discussion since I wrote this code. Yes, we need to change this thing! I think instead of specifying the auth type like this we should have just a flag indicating if the auth is enabled or not. (because in this PR we assume that only desc auth is available)